.gitignore

```diff
@@ -2,6 +2,8 @@ _esy
 _build
 _public
 .merlin
+esy.lock
```

Comment:
I'd like to start building `httpkit` with esy sometime soon :D Can we drop the changes to the `.gitignore`?

Reply:
I'm currently using `esy` actually but didn't want to commit my `package.json` and locks, and mix it with this PR.

I'll open a separate PR if that's ok with you